### DISCUSSION TITLE
Mainline kernel testing and zfs

### DISCUSSION THREAD

**Discussion by gc-user** [void-linux/void-packages#52150] [Q&A]
I was wondering what's the worst that could happen if I wanted to test mainline kernel on a zfs system.

Kernel bugs apparently have to be tested on / against mainline.
zfs is not (officially) supporting mainline kernel, but is at least one version behind in kernel support.

Now, I could set the maximum version for zfs to mainline version and compile it that way.

So, I'm wondering what's the worst that could happen? Could booting into mainline kernel with zfs destroy / harm the whole tank / zpool or is the damage always only contained to the boot environment? The latter would be ok, because it would be a test BE anyway.

I guess, something not working quite right with such a setup would have to be expected and "acceptable" (for testing), but doing any harm to the outside of the test BE (i.e. the zpool in general) would be a no-no. That's why I'm asking.

**Comment by classabbyamp** [comment]
> what's the worst that could happen?

catastrophic data loss from a bug

**Reply by gc-user** [reply]
Thanks for your reply, I'm aware of that, but is that data loss possible only within the BE or also in the zpool that contains the BE - that's my question.

**Reply by classabbyamp** [reply]
either

**Reply by gc-user** [reply]
That's not what I was hoping for, but was thinking, too... :-)

Well, then it's  settled: Use a different drive for testing... maybe with bcachfs instead of zfs... :-)
